src/main/java/seedu/address/logic/parser/EditQuestionCommandParser.java

```diff
@@ -23,14 +20,21 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public EditQuestionCommand parse(String arguments) throws ParseException {
-        Matcher matcher = BASIC_EDIT_QUESTION_COMMAND_FORMAT.matcher(arguments.trim());
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(arguments, PREFIX_QUESTION);
 
-        if (!matcher.matches()) {
+        if (!argMultimap.arePrefixesPresent(PREFIX_QUESTION)) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditQuestionCommand.MESSAGE_USAGE));
         }
-        final String questionNumber = matcher.group("questionNumber");
-        final String newQuestion = matcher.group("newQuestion");
-
-        return new EditQuestionCommand(questionNumber.trim(), newQuestion.trim());
+        if (argMultimap.getValue(PREFIX_QUESTION).get().equals("")) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditQuestionCommand.MESSAGE_USAGE));
+        }
+        try {
+            int index = Integer.parseInt(argMultimap.getPreamble());
+            return new EditQuestionCommand(index, argMultimap.getValue(PREFIX_QUESTION).get());
+        } catch (NumberFormatException e) {
+            throw new ParseException(INDEX_NOT_A_NUMBER);
+        }
```

Comment:
Same, to make it consistent, is it better to use the MESSAGE_UNKNOWN_COMMAND instead?